### PR TITLE
implement kubevirt format

### PR DESCRIPTION
**Pull request by jbpratt**
requires https://github.com/NixOS/nixpkgs/pull/151482

Signed-off-by: jbpratt <jbpratt78@gmail.com>

### CODE REVIEW

**Comment by jbpratt** [comment]
Hi @Lassulus! Thank you for your work on `nixos-generate`. I'm working on adding basic support to build KubeVirt disk images, but I'm incredibly new to Nix and the ecosystem, so my apologies in advance.

I initially attempted using the `openstack` and `cloudstack` formats but `openstack` doesn't work since it sets up the metadata fetcher not needed here, and is missing cloud-init; `cloudstack` sets up an additional data source and a package. It is also not working at the moment after https://github.com/NixOS/nixpkgs/pull/131760 (I believe, since it is missing `fsType`).

I'm still working out the kinks but wanted to get your feedback on both this and the upstream PR.

File: formats/kubevirt.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ config, lib, pkgs, modulesPath, ... }:
+{
+  imports = [
+    "${toString modulesPath}/virtualisation/kubevirt-config.nix"
+  ];
+
+  system.build.kubevirtImage =  import "${toString modulesPath}/../lib/make-disk-image.nix" {
+    inherit lib config pkgs;
+    diskSize = 8192;
```
Comment (Lassulus) [inline formats/kubevirt.nix:9]: if you leave the diskSize empty it should automatically detect the required space

Comment (jbpratt) [reply formats/kubevirt.nix]: removed!

**Comment by Mic92** [comment]
I just temporary close & open this, so that I don't have to approve tests each time.

**Pull request merged**